tests/letstest/scripts/test_leauto_upgrades.sh

```diff
@@ -15,19 +15,54 @@ if ! command -v git ; then
         exit 1
     fi
 fi
-BRANCH=`git rev-parse --abbrev-ref HEAD`
 # 0.5.0 is the oldest version of letsencrypt-auto that can be used because it's
 # the first version that pins package versions, properly supports
 # --no-self-upgrade, and works with newer versions of pip.
-git checkout -f v0.5.0
+git checkout -f v0.5.0 letsencrypt-auto
 if ! ./letsencrypt-auto -v --debug --version --no-self-upgrade 2>&1 | grep 0.5.0 ; then
     echo initial installation appeared to fail
     exit 1
 fi
 
-git checkout -f "$BRANCH"
-EXPECTED_VERSION=$(grep -m1 LE_AUTO_VERSION letsencrypt-auto | cut -d\" -f2)
-if ! ./letsencrypt-auto -v --debug --version --no-self-upgrade 2>&1 | grep $EXPECTED_VERSION ; then
+# Now that python and openssl have been installed, we can set up a fake server
+# to provide a new version of letsencrypt-auto. First, we start the server and
+# directory to be served.
+MY_TEMP_DIR=$(mktemp -d)
+SERVER_PORT=12345
```

Comment:
Should do something like https://github.com/certbot/certbot/blob/master/letsencrypt-auto-source/tests/auto_test.py#L84 if we don't want this test to be flaky.

Reply:
Or, if SimpleHTTPServer can get a port for you and tell you what it is.

Reply:
If you tell `SimpleHTTPServer` to use port 0, the OS gives it a random free port. I had a version that did this but encountered some issues where Python wasn't flushing the output stream when I had it redirected somewhere, causing it to never print the port it chose.

I can either add a simple recreation of this server to `tools/` that manually flushes the output stream or we can pick a random, unprivileged port like 12345 (or another number) and assume/hope there won't be any collisions on our simple EC2 instances. Preference?

Reply:
Any reason not to do what https://github.com/certbot/certbot/blob/master/letsencrypt-auto-source/tests/auto_test.py#L84 does? Pick a range of ports, and try ports in that range until it works.

Reply:
I think there's an awkward race condition. We start `SimpleHTTPServer` in the backgrpound and since it's running in parallel, we're just have to pick an arbitrary amount of time to wait and see if it's going to fail. Also, I'm not aware of a good way to check the exit status of a background process (without blocking on the process exiting).

Reply:
I put a version that has the OS pick the port in #5390. Pick whichever you like better or let me know if there's another approach you prefer that I'm missing.